### PR TITLE
Chore/jspc compile fixes migration

### PR DESCRIPTION
**Pull request by yingbull**
This pull request includes several improvements and refactorings across multiple JSP files, focusing on code quality, maintainability, and user experience. The main themes are robust handling of nulls and edge cases, improved event listener attachment, consistent file includes, and code cleanup for better readability and maintainability.

**Chart Notes Handling and JavaScript Refactoring:**

* Improved null safety and logging for `notesToDisplay`, preventing possible null pointer exceptions and adding a warning log if the attribute is missing. The logic for note rendering and event listener attachment was refactored for clarity and to ensure listeners are attached only once, even on paginated loads. User messaging and note editing logic were also streamlined for better maintainability and user experience. [[1]](diffhunk://#diff-748df8d3f23ca8af8813918f44358b88cf6a072d8d46695b253e8398a89f3125L148-R151) [[2]](diffhunk://#diff-748df8d3f23ca8af8813918f44358b88cf6a072d8d46695b253e8398a89f3125L175-R178) [[3]](diffhunk://#diff-748df8d3f23ca8af8813918f44358b88cf6a072d8d46695b253e8398a89f3125L795-R806) [[4]](diffhunk://#diff-748df8d3f23ca8af8813918f44358b88cf6a072d8d46695b253e8398a89f3125L859-R904) [[5]](diffhunk://#diff-748df8d3f23ca8af8813918f44358b88cf6a072d8d46695b253e8398a89f3125L894-L906) [[6]](diffhunk://#diff-748df8d3f23ca8af8813918f44358b88cf6a072d8d46695b253e8398a89f3125L922-R957) [[7]](diffhunk://#diff-748df8d3f23ca8af8813918f44358b88cf6a072d8d46695b253e8398a89f3125L959-R984) [[8]](diffhunk://#diff-748df8d3f23ca8af8813918f44358b88cf6a072d8d46695b253e8398a89f3125L1012-L1015)

**File Include Path Consistency:**

* Updated JSP include paths for `global-head.jspf` to use the correct, consistent location (`/WEB-INF/jsp/includes/global-head.jspf`) across multiple files, improving maintainability and reducing the risk of broken includes. [[1]](diffhunk://#diff-5345888bb28f63c2b1c64702f1bf50acf897ebb97ecb1901b2750bd77fb07671L39-R39) [[2]](diffhunk://#diff-47152cda2b569d671a5850797cba2faf4e5d7ba903a242f9793ed91a67787dd2L48-R48) [[3]](diffhunk://#diff-c7fd0181c5d724480f46f8204d5767378efabfdf9816b9fb4bc35f60c9db768dL39-R39) [[4]](diffhunk://#diff-f349a122cba2f6cd3bc766d9e7b15352f5596fe6e3f497a58765f373079aadddL39-R39) [[5]](diffhunk://#diff-5f296f02bce9d4172ae399b3550cffc9baf4a5428391cac24110953cd605c89eL39-R39) [[6]](diffhunk://#diff-a3b6c35c87d5b9519057f317b904b4f1943dbfa15f92ddee73967c41353d0ce3L21-R21)

**Encoding and String Handling Improvements:**

* Switched from `SafeEncode.forJavaScript` to `SafeEncode.forJavaScriptAttribute` in address handling to ensure proper encoding for JavaScript attributes.
* Refactored string processing in lab and provider encounter print JSPs to improve clarity and efficiency, including precomputing normalized and preview notes and extracting repeated logic into variables. [[1]](diffhunk://#diff-9c2f60b2bac806857e7f3a61bfc042fb49d7dee1eab073804fc8fd2f2c232290R304-R310) [[2]](diffhunk://#diff-84f4c17bfaaade73a2f129c4b2fc439a8cc8e59aa6e600da7ecbb1bb62aeb17fR145-R150)

**Taglib and Import Cleanup:**

* Moved and deduplicated taglib declarations in `formBCAR2020pg1.jsp` for clarity and to prevent redundant imports. [[1]](diffhunk://#diff-9a1eeeb2836ca35aeae6a511c41b23901a81ecf14524913fa223c77a4a787517R58) [[2]](diffhunk://#diff-9a1eeeb2836ca35aeae6a511c41b23901a81ecf14524913fa223c77a4a787517L185-L186)

These changes collectively improve code reliability, maintainability, and user experience across the application.

## Summary by Sourcery

Improve chart notes rendering robustness and client-side behavior while standardizing JSP utilities and includes.

Bug Fixes:
- Prevent null pointer issues and missing-attribute failures when rendering chart notes.
- Ensure fullView click listeners are attached reliably for unlocked notes across initial and paginated loads.
- Correct JavaScript encoding for third-party billing address values used in JS attributes.

Enhancements:
- Streamline chart note display and locking logic, including single-line formatting, oscar messages, and new vs existing note handling.
- Refine lab report and encounter print JSPs by precomputing normalized and preview strings to simplify markup and improve readability.
- Deduplicate and centralize the carlos taglib declaration in the BCAR 2020 form JSP for cleaner imports.
- Standardize global-head JSP include paths across multiple pages to use the shared WEB-INF include.

---
## Summary by cubic
Make JSPC compile cleanly and harden chart notes rendering and scripting. Standardizes include paths and encoding, and fixes DOM nesting and null handling across JSPs.

- **Bug Fixes**
  - Chart notes: close per-note wrapper divs to restore valid DOM and click targeting; null-guard `casemgmtNoteLock`/`oscarMsg`; attach `fullView` listeners once per render.
  - Prevent NPEs by checking `notesToDisplay` and logging when missing.
  - Guard provider encounter reason before `substring(2)`; normalize and truncate lab OBX notes safely to avoid partial `\.br\` artifacts.
  - Use `SafeEncode.forJavaScriptAttribute` for `onClick` values in BC 3rd‑party billing search.
  - Use consistent include path `/WEB-INF/jsp/includes/global-head.jspf` across pages.

- **Refactors**
  - Streamlined chart notes init and lock flow; fewer inline scripts and clearer flags.
  - Deduplicated taglib declarations in `formBCAR2020pg1.jsp`.

<sup>Written for commit 4ae39f9def2c74a64dd0613bc01bbb8a12582b6f. Summary will update on new commits.</sup>

### CODE REVIEW

File: src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp [hunk 1]
```diff
@@ -919,69 +937,51 @@ EmailComposeManager emailComposeManager = SpringUtils.getBean(EmailComposeManage
...
-		// Let the paste happen first, then resize
-		setTimeout(adjustCaseNote, 0);
-	});
+    jQuery('#' + caseNote).on('keyup', monitorCaseNote);
+    jQuery('#' + caseNote).on('paste', function(e) {
+        // Let the paste happen first, then resize
+        setTimeout(adjustCaseNote, 0);
+    });
 
     document.getElementById(caseNote).addEventListener('click', getActiveText);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp:987]: Acknowledged but not changing. The `caseManagementEntryForm` form lives in `newCaseManagementView.jsp` which is rendered before this AJAX fragment loads, so it's guaranteed present when the script runs. The script also already has selective null checks where they matter (e.g. line 896 `if (document.getElementById(caseNote))`). Adding broader null checks would diverge from the established pattern in this file.

File: src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp [hunk 2]
```diff
@@ -919,69 +937,51 @@ EmailComposeManager emailComposeManager = SpringUtils.getBean(EmailComposeManage
...
-             {%>
-    document.getElementById(caseNote).value += "\n\n<carlos:encode value='<%= bean.oscarMsg %>' context="javaScript"/>";
-    <%bean.reason = "";
-                 bean.oscarMsg = "";
-             }
+    // Existing note: flag note_edit as "existing". New note: init newNoteIdx.
+    if (<%= found %>) {
+        document.forms["caseManagementEntryForm"].note_edit.value = "existing";
+    } else {
+        document.forms["caseManagementEntryForm"].newNoteIdx.value = <%=savedId%>;
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp:977]: Not applicable here. `savedId` is declared `long savedId = 0;` at line 142 — it's always a numeric primitive, never user-controlled or string-typed. Adding quotes would change the JavaScript type from `Number` to `String` and break downstream numeric comparisons. The unquoted form is correct.

File: src/main/webapp/WEB-INF/jsp/lab/CA/BC/report.jsp [hunk 3]
```diff
@@ -301,9 +301,13 @@
             </td>
             <td class="Text" nowrap class="<%=(other? "LightBG" : "WhiteBG")%>"><carlos:encode value='<%= hl7_obx.getUnits() %>' context="html"/>
             </td>
+            <%
+                String normalizedObxNote = hl7_obx.getNote().replace("\\.br\\", " ");
+                String previewObxNote = (hl7_obx.getNote().length() < 20 ? hl7_obx.getNote() : hl7_obx.getNote().substring(0, 20)).replace("\\.br\\", " ");
+            %>
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/lab/CA/BC/report.jsp:310]: Fixed in 4ae39f9. Hoisted to a `rawObxNote` local with empty-string fallback. Verified `Hl7Obx.getNote()` is a vanilla JPA getter with no internal null safety, so the guard was needed.

File: src/main/webapp/WEB-INF/jsp/provider/providerencounterprint.jsp [hunk 4]
```diff
@@ -142,9 +142,12 @@
 
                 <table width="100%" cellspacing="0" cellpadding="2" border="1"
                        datasrc='#xml_list'>
+                    <%
+                        String encounterReason = subject.substring(2).replace("|", " ");
+                    %>
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/provider/providerencounterprint.jsp:149]: Fixed in 4ae39f9. Added `subject != null && subject.length() >= 2` guard with empty-string fallback before calling `substring(2)`. `subject` can indeed be null when `enc == null` per the init at line 78.

File: src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp [hunk 5]
```diff
@@ -782,42 +777,33 @@ EmailComposeManager emailComposeManager = SpringUtils.getBean(EmailComposeManage
...
-            if (!fulltxt && !note.isDocument() && !note.isEformData() && !note.isEncounterForm() && !note.isRxAnnotation() && !note.isInvoice() && !note.isEmailNote())
-            {
-                %><script> document.getElementById('n<%=note.getNoteId()%>').addEventListener('click', fullView); </script><%
-                unLockedNotes.add(note.getNoteId());
-            }
-
-        } //end for */
-    %>
-</c:if> <%-- END OF "not empty notesToDisplay" --%>
+    <!-- end of note wrapper -->
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp:817]: Fixed in 4ae39f9 — applied your suggested patch. Verified via a brace tracer: the for-loop now closes at line 819 (after the `</div>` tags), so each iteration emits its own wrapper closes. Thanks for catching this — the original brace imbalance (108 opens vs 107 closes overall) was the underlying JSPC failure this PR was created to fix, and removing the c:if wrapper exposed the misplaced for-close.

File: src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp [hunk 6]
```diff
@@ -856,54 +842,86 @@ EmailComposeManager emailComposeManager = SpringUtils.getBean(EmailComposeManage
...
+    String singleLineFormat = "false";
+    if (renderMoreNotesScript) {
+        UserProperty slProp = (UserProperty) request.getAttribute(UserProperty.STALE_FORMAT);
+        if (slProp != null && slProp.getValue().equals("yes")) {
+            singleLineFormat = "true";
+        }
+    }
+
+    String oscarMsg = bean.oscarMsg;
+    boolean hasOscarMsg = renderMoreNotesScript && !oscarMsg.equals("");
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp:879]: Fixed in 4ae39f9. Introduced `hasNoteLock = casemgmtNoteLock != null`, gated `noteLocked`/`noteLockedBySameUser` on it, switched session-id comparison to `Objects.equals(...)`, and defaulted `bean.oscarMsg` to `""` before `.equals("")`. Added `java.util.Objects` import.

File: src/main/webapp/WEB-INF/jsp/lab/CA/BC/report.jsp [hunk 7]
```diff
@@ -301,9 +301,13 @@
...
             <td class="Text" nowrap
-                title="<carlos:encode value='<%= hl7_obx.getNote().replaceAll("\\\\\\.br\\\\", " ") %>' context="htmlAttribute"/>"
-                class="<%=(other? "LightBG" : "WhiteBG")%>"><carlos:encode value='<%= ((hl7_obx.getNote().length() < 20) ? hl7_obx.getNote() : hl7_obx.getNote().substring(0, 20)).replaceAll("\\\\\\.br\\\\", " ") %>' context="html"/>
+                title="<carlos:encode value='<%= normalizedObxNote %>' context="htmlAttribute"/>"
+                class="<%=(other? "LightBG" : "WhiteBG")%>"><carlos:encode value='<%= previewObxNote %>' context="html"/>
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/lab/CA/BC/report.jsp:313]: Fixed in 4ae39f9, combined with the `getNote()` null-guard from the other report.jsp thread — now: cache → normalize → truncate. Re your note that `getNote()` is null-safe: I checked `Hl7Obx.getNote()` and it's a vanilla JPA getter (`return note;`) with no null safety, so I kept the explicit fallback.

File: src/main/webapp/WEB-INF/jsp/provider/providerencounterprint.jsp [hunk 8]
```diff
@@ -142,9 +142,12 @@
...
                     <tr>
                         <td><carlos:encode value='<%= encounter_date %>' context="html"/> <carlos:encode value='<%= encounter_time %>' context="html"/><br>
-                            <b><fmt:message key="provider.providerencounterprint.reason"/>:</b><carlos:encode value='<%= subject.substring(2).replaceAll("\\|", " ") %>' context="html"/><br>
+                            <b><fmt:message key="provider.providerencounterprint.reason"/>:</b><carlos:encode value='<%= encounterReason %>' context="html"/><br>
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/provider/providerencounterprint.jsp:152]: Fixed in 4ae39f9 with the suggested guard.

File: src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp [hunk 9]
```diff
@@ -145,7 +145,10 @@ EmailComposeManager emailComposeManager = SpringUtils.getBean(EmailComposeManage
 
     @SuppressWarnings("unchecked")
     ArrayList<NoteDisplay> notesToDisplay = (ArrayList<NoteDisplay>) request.getAttribute("notesToDisplay");
-    int noteSize = notesToDisplay.size();
+    if (notesToDisplay == null) {
+        MiscUtils.getLogger().warn("notesToDisplay request attribute missing for demographic " + demographicNo);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/casemgmt/ChartNotesAjax.jsp]: Already addressed in an earlier commit on this branch. The log statement now uses `LogSanitizer.sanitize(demographicNo)` (prevents log injection / CRLF forging) and the level was raised to ERROR — this is a programming defect (every supported entry path sets the attribute), not a recoverable WARN. Per `LogSanitizer` JavaDoc, internal IDs like demographic numbers are appropriate to log; PHI policy applies to names/HIN/DOB/clinical data. The empty-render now also shows a visible alert and returns early so the clinician doesn't mistake a broken render for an empty chart.

File: src/main/webapp/WEB-INF/jsp/provider/providerencounterprint.jsp [hunk 10]
```diff
@@ -142,9 +142,12 @@
 
                 <table width="100%" cellspacing="0" cellpadding="2" border="1"
                        datasrc='#xml_list'>
+                    <%
+                        String encounterReason = subject.substring(2).replace("|", " ");
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/provider/providerencounterprint.jsp]: Fixed in 4ae39f9 with a `subject != null && subject.length() >= 2` guard and empty-string fallback.

File: src/main/webapp/WEB-INF/jsp/lab/CA/BC/report.jsp [hunk 11]
```diff
@@ -301,9 +301,13 @@
             </td>
             <td class="Text" nowrap class="<%=(other? "LightBG" : "WhiteBG")%>"><carlos:encode value='<%= hl7_obx.getUnits() %>' context="html"/>
             </td>
+            <%
+                String normalizedObxNote = hl7_obx.getNote().replace("\\.br\\", " ");
+                String previewObxNote = (hl7_obx.getNote().length() < 20 ? hl7_obx.getNote() : hl7_obx.getNote().substring(0, 20)).replace("\\.br\\", " ");
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/lab/CA/BC/report.jsp]: Fixed in 4ae39f9 — cached `getNote()` to a local with empty-string fallback before calling `replace`/`length`/`substring`.

File: src/main/webapp/WEB-INF/jsp/provider/providerencounterprint.jsp (hunk 10 above)
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/provider/providerencounterprint.jsp]: Fixed in 4ae39f9 with the suggested null/length guard.
